src/ReplicateAdapter.php

```diff
@@ -252,4 +256,20 @@ public function setVisibility($path, $visibility)
 
         return $this->replica->setVisibility($path, $visibility);
     }
-}
+
+    /**
+     * Return resource or binary stream that can using for `writeStream` and
+     * `updateStream`.
+     *
+     * @param mixed $resource
+     * @param string $path
+     * @return mixed
+     */
+    protected function ensureSeekable($resource, $path)
+    {
+        if (is_resource($resource) && fseek($resource, 0) !== 0) {
```

Comment:
No need to check for is_resource() here.

This should use Util::isSeekableStream() and either rewind or return the read stream.